metapool/literals.py

```diff
@@ -0,0 +1,112 @@
+# them are really cross-domain and which really do belong to a specific
+# domain (where they can then be moved in the future).  Constants that I
+# already *know* to be domain-specific are not included here.
+
+# Note:
+# A number of these string literals are also used in amplicon-related code
+# (e.g., prep.py, scripts/seqpro.py, seqpro_mf.py, etc.).  However, I don't
+# know if they are the same because they come from the same source, or if it is
+# just coincidental.  For now, I have NOT changed the amplicon-related
+# instances to use these constants.
+
+import re
+
+# These are the "standard"-format snake-case column names;
+# if writing something NEW, use these
+SAMPLE_NAME_KEY = "sample_name"
+QIITA_ID_KEY = "qiita_id"
+SAMPLE_PROJECT_KEY = "sample_project"
+ORIG_NAME_KEY = "orig_name"
+EXPT_DESIGN_DESC_KEY = 'experiment_design_description'
+CONTAINS_REPLICATES_KEY = "contains_replicates"
+PROJECT_SHORT_NAME_KEY = "short_project_name"
+PROJECT_FULL_NAME_KEY = "full_project_name"
+# NB: don't use this as the name for a column of sample identifiers.  It is
+# the key for a list of sample info *objects* (e.g., dictionaries)
+SAMPLES_KEY = "samples"
+
+# Plate map (PM) column names
+PM_SAMPLE_KEY = "Sample"
+PM_PROJECT_PLATE_KEY = "Project Plate"
+PM_PROJECT_NAME_KEY = "Project Name"
+PM_PROJECT_ABBREV_KEY = 'Project Abbreviation'
+PM_COMPRESSED_PLATE_NAME_KEY = "Compressed Plate Name"
+PM_BLANK_KEY = "Blank"
+PLATE_NAME_DELIMITER = "_"
+
+# Currently used to describe sample context (only)
+SAMPLE_TYPE_KEY = "sample_type"
+PRIMARY_STUDY_KEY = "primary_qiita_study"
+SECONDARY_STUDIES_KEY = "secondary_qiita_studies"
+
+
+def get_plate_name_delimiter():
+    return PLATE_NAME_DELIMITER
+
+
+def parse_project_name(project_name):
+    """
+    Split fully-qualified project_name into a short project_name and a qiita-id
+    :param project_name: A fully-qualified project name e.g: Feist_1161.
+    :return: Dictionary of qiita_id, short_project_name, and full_project_name
+    """
+    if project_name is None or project_name == '':
+        raise ValueError("project_name cannot be None or empty string")
+
+    # project identifiers are digit groups at the end of the project name
+    # preceded by an underscore CaporasoIllumina_550
+    matches = re.search(r'^(.+)_(\d+)$', str(project_name))
+
+    if matches is None:
+        raise ValueError(f"'{project_name}' does not contain a Qiita-ID.")
+
+    proj_info_dict = {
+        QIITA_ID_KEY: matches[2],
+        PROJECT_SHORT_NAME_KEY: matches[1],
+        PROJECT_FULL_NAME_KEY: project_name
+    }
+
+    return proj_info_dict
+
+
+def get_short_name_and_id(project_name):
+    try:
+        proj_info_dict = parse_project_name(project_name)
+    except ValueError:
+        return project_name, None
+
+    return proj_info_dict[PROJECT_SHORT_NAME_KEY], proj_info_dict[QIITA_ID_KEY]
+
+
+def get_qiita_id_from_project_name(project_name):
```

Comment:
Same here, why a function?

Reply:
Because calling this function replaces doing in-the-weeds string manipulation *all over* this code base, which is a very brittle idiom.  Right now functions spread across three repositories have baked-in knowledge of the string format of project names, and this means that if we ever need to change anything about that format--or even if we discover we need to be more careful in interpreting the *existing* format (as has happened more than once with the string handling used to parse fastq file names), we have a big, fiddly task to find and modify all those functions (and we're pretty likely to miss one somewhere ... ).  We should take a lesson from the situation with the blanks, where the identification of blanks via string matching was happening separately in five places across three repos--but was only the same in *four* of them, and the outlier was incompatible with the new NPH project.

Reply:
Sounds good, thank you for the explanation. I guess this highlights the "advantages" of OOP, where we could have a class "Project", then a property/method `name` and another `qiita_id`, which are only parsed once and used along the code and inherited by other software/code vs. multiple smaller functions. Note that I'm not suggesting to make this change.

Reply:
one of the few yes? 😄